drivers/hwmon_ina219.c

```diff
@@ -0,0 +1,496 @@
+/* hwmon.c Driver for INA219 hwmon-based power monitors.
+
+	Copyright (C) 2024 Jan Viktorin <jan.viktorin@gmail.com>
+
+	This program is free software; you can redistribute it and/or modify
+	it under the terms of the GNU General Public License as published by
+	the Free Software Foundation; either version 2 of the License, or
+	(at your option) any later version.
+
+	This program is distributed in the hope that it will be useful,
+	but WITHOUT ANY WARRANTY; without even the implied warranty of
+	MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+	GNU General Public License for more details.
+
+	You should have received a copy of the GNU General Public License
+	along with this program; if not, write to the Free Software
+	Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA 02111-1307 USA
+ */
+
+#include "main.h"
+#include "nut_float.h"
+#include "nut_stdint.h"
+
+#include <dirent.h>
+#include <errno.h>
+#include <string.h>
+#include <sys/types.h>
+
+#ifndef STRFY0
+	#define STRFY0(x) #x
+#endif
+#ifndef STRFY
+	#define STRFY(x) STRFY0(x)
+#endif
+
+#define SYSFS_HWMON_DIR                     "/sys/class/hwmon"
+#define BATTERY_CHARGE_LOW                  15
+#define BATTERY_CHARGE_HIGH                 75
+#define DRIVER_NAME                         "hwmon-INA219 UPS driver"
+#define DRIVER_VERSION                      "0.01"
+
+upsdrv_info_t upsdrv_info = {
+	DRIVER_NAME,
+	DRIVER_VERSION,
+	"Jan Viktorin <jan.viktorin@gmail.com>",
+	DRV_EXPERIMENTAL,
+	{ NULL },
+};
+
+/**
+ * @brief Path usually pointing to /sys/class/hwmon/hwmonX.
+ */
+static char ina219_base_path[PATH_MAX];
+
+/**
+ * @brief Threshold for detection of LB status.
+ */
+static unsigned int battery_charge_low = BATTERY_CHARGE_LOW;
```

Comment:
Hm, per `docs/nut-names.dict` this could in fact be set as `default.battery.charge.low` (instead of a custom `lb` parameter), and in fact there currently is no reserved name for the `.high` but makes sense to add one. I only see a handful of precedents setting the `HB` flag so had to wonder a bit if it is "just informative" or a warning of potential overcharge (harmful to the device). Would ask the mailing list just in case.

Reply:
I think it might, just not sure about the history behind the flag - too few hits in code to make sense of it quickly - hence why I'll ask on the ML.

For that matter, I see a couple of hits to `git grep -iE 'over.*charge'` that refer to over-charged battery state (and nearby some references to over-temperature flags), and they are in different sources compared to `git grep -w HB`.

Reply:
I based my high-battery threshold assumption on `pijuice.c`:

```
413     if ( batteryStatus == BATT_NORMAL ||                                         
414          batteryStatus == BATT_CHARGING_FROM_IN ||                               
415          batteryStatus == BATT_CHARGING_FROM_5V )                                
416     {                                                                            
417         get_charge_level_hi_res();                                               
418                                                                                  
419         if ( battery_charge_level <= LOW_BATTERY_THRESHOLD )                     
420         {                                                                        
421             upsdebugx( 1, "Battery Charge Status: LOW" );                        
422             snprintfcat( status_buf, ST_MAX_VALUE_LEN, "LB " );                  
423         }                                                                        
424         else if ( battery_charge_level > HIGH_BATTERY_THRESHOLD )                
425         {                                                                        
426             upsdebugx( 1, "Battery Charge Status: HIGH" );                       
427             snprintfcat( status_buf, ST_MAX_VALUE_LEN, "HB " );                  
428         }                                                                        
429     }
```

I removed the `hb` option and changed `lb` to `battery.charge.low`.